### ISSUE TITLE
Remote export to windows stuck on "Creating temporary directory"

### ISSUE THREAD

**Issue by eh-jogos** [godotengine/godot#73965]
### Godot version

4.0rc4

### System information

Manjaro Linux using i3, Foward+

### Issue description

I setup ssh server between my linux machine and my windows machine, through the terminal I can connect from windows to linux no problems, using ssh key and no passwords.

Then I tried to turn on remote export for windows.

I've configured the windows export preset with my host and port and left everything else on default
![image](https://user-images.githubusercontent.com/13070158/221397368-ec5328ca-582a-4769-8da5-e62f427a6c91.png)

When I run the remote export I get stuck on "Creating temporary directory" forever.
On the terminal I get the following messages 
![image](https://user-images.githubusercontent.com/13070158/221397441-54b57864-4f69-4550-bc16-d85e7847a803.png)

At this point I either have to close the terminal and let godot crash or forcibly close godot either with htop or xkill. I don't know why its failing and I get no message to help me fix it.

### Steps to reproduce

Configure ssh server to be able to connect to a windows 11 machine from a linux machine
Try to remote export a project to it

### Minimal reproduction project

Steps above?

**Comment by akien-mga** [comment]
Duplicate of https://github.com/godotengine/godot/issues/73842

**Comment by eh-jogos** [comment]
Only once while testing this with godot 4.0 stable I got a new message:
```
Creating temporary directory...
= : The term '=' is not recognized as the name of a cmdlet, function, script file, or operable 
program. Check the spelling of the name, or if a path was included, verify that the path is 
correct and try again.
At line:1 char:2
+  = Join-Path C:\Users\danie\AppData\Local\Temp 8c8987ad-b057-4325-b4f ...
+  ~
    + CategoryInfo          : ObjectNotFound: (=:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException
 
New-Item : Missing an argument for parameter 'Path'. Specify a parameter of type 
'System.String[]' and try again.
At line:1 char:111
+ ... d-b057-4325-b4fe-d0382b40a85c; New-Item -Type Directory -Path  | Out- ...
+                                                             ~~~~~
    + CategoryInfo          : InvalidArgument: (:) [New-Item], ParameterBindingException
    + FullyQualifiedErrorId : MissingArgument,Microsoft.PowerShell.Commands.NewItemCommand
 
Write-Output : Cannot process command because of one or more missing mandatory parameters: 
InputObject.
At line:1 char:130
+ ... 0382b40a85c; New-Item -Type Directory -Path  | Out-Null; Write-Output
+                                                              ~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Write-Output], ParameterBindingException
    + FullyQualifiedErrorId : MissingMandatoryParameter,Microsoft.PowerShell.Commands.WriteOutput 
   Command
 

xarchiver 0.5.4 ©  2005-2014 Giuseppe Torelli
xarchiver 0.5.4.20 Ingo Brückl

```

I saw there were some fixes on #74030 and I was hoping this might have helped:
> Fixes Windows remote execute failing with empty arguments.

So I tested it with 4.0.1.rc1 and still stuck on "Creating temporary directory..."

Tried adding a `-i` argument to the export preset with the path to my key to not leave it blank, even though I don't need it when using `ssh` in the terminal, and both 4.0.stable and 4.0.1.rc1 got stuck on "Creating temporary directory..." without giving any other errors.

**Comment by eh-jogos** [comment]
also tried to add `-v` both to ssh extra args and scp extra args and there was no prints from them running in the console, nothing after "Creating temporary directory...", so I guess they're not even running?

**Comment by azuriancomet** [comment]
> Only once while testing this with godot 4.0 stable I got a new message:
> 
> ```
> Creating temporary directory...
> = : The term '=' is not recognized as the name of a cmdlet, function, script file, or operable 
> program. Check the spelling of the name, or if a path was included, verify that the path is 
> correct and try again.
> At line:1 char:2
> +  = Join-Path C:\Users\danie\AppData\Local\Temp 8c8987ad-b057-4325-b4f ...
> +  ~
>     + CategoryInfo          : ObjectNotFound: (=:String) [], CommandNotFoundException
>     + FullyQualifiedErrorId : CommandNotFoundException
>  
> New-Item : Missing an argument for parameter 'Path'. Specify a parameter of type 
> 'System.String[]' and try again.
> At line:1 char:111
> + ... d-b057-4325-b4fe-d0382b40a85c; New-Item -Type Directory -Path  | Out- ...
> +                                                             ~~~~~
>     + CategoryInfo          : InvalidArgument: (:) [New-Item], ParameterBindingException
>     + FullyQualifiedErrorId : MissingArgument,Microsoft.PowerShell.Commands.NewItemCommand
>  
> Write-Output : Cannot process command because of one or more missing mandatory parameters: 
> InputObject.
> At line:1 char:130
> + ... 0382b40a85c; New-Item -Type Directory -Path  | Out-Null; Write-Output
> +                                                              ~~~~~~~~~~~~
>     + CategoryInfo          : InvalidArgument: (:) [Write-Output], ParameterBindingException
>     + FullyQualifiedErrorId : MissingMandatoryParameter,Microsoft.PowerShell.Commands.WriteOutput 
>    Command
>  
> 
> xarchiver 0.5.4 ©  2005-2014 Giuseppe Torelli
> xarchiver 0.5.4.20 Ingo Brückl
> ```
> 
> I saw there were some fixes on [#74030](https://github.com/godotengine/godot/pull/74030) and I was hoping this might have helped:
> 
> > Fixes Windows remote execute failing with empty arguments.
> 
> So I tested it with 4.0.1.rc1 and still stuck on "Creating temporary directory..."
> 
> Tried adding a `-i` argument to the export preset with the path to my key to not leave it blank, even though I don't need it when using `ssh` in the terminal, and both 4.0.stable and 4.0.1.rc1 got stuck on "Creating temporary directory..." without giving any other errors.

Got the exact same thing v4.4.1.stable.fedora [49a5bc7b6].
I'm guessing the PowerShell script is somehow broken: https://github.com/godotengine/godot/blob/master/platform/windows/export/export_plugin.cpp#L1117
That's where it stops and prints the error. I've got no idea what it could be since I don't speak PowerShell, and I'm all out of ideas.

**Comment by vilhalmer** [comment]
A hint for others trying to get this working when the above workarounds don't help:

I ran the editor in verbose mode and saw that ssh itself was failing (exit code 255):
```
Creating temporary directory...
Executing: ssh -p 22 -q -o LogLevel=error -o BatchMode=yes -o StrictHostKeyChecking=no me@windows.local powershell -command \"\$tmp = Join-Path \$Env:Temp \$(New-Guid); New-Item -Type Directory -Path \$tmp | Out-Null; Write-Output \$tmp\"
Exit code: 255
```

If I ran the ssh command manually without `-q` it showed that it was trying to use password auth, meaning that my ssh key wasn't being accepted. This turns out to be due to the default sshd_config on Windows 10: https://superuser.com/a/1651276

> Here's the most likely cause ... If your Windows' user is an administrator, then the default Windows OpenSSH configuration is probably "getting in the way" of our sensible Unix/Linux expectations ;-). Check your C:\ProgramData\ssh\sshd_config. By default, it probably has the following lines at the bottom:

```
Match Group administrators
    AuthorizedKeysFile __PROGRAMDATA__/ssh/administrators_authorized_keys
```

I commented out these lines to allow the normal `%UserProfile%/.ssh/authorized_keys` file to work, though you could opt to put your key in the referenced file instead. (Also note I had no luck getting `ssh-copy-id` to install the key into that path from a linux machine, and just did it by hand.)

If this seems reasonable to add to the troubleshooting section of the One-click deploy documentation, I can work on a PR.
